From: Catherine Sullivan <catherine.sullivan@intel.com>
Date: Wed, 4 Jun 2014 08:45:24 +0000
Subject: [PATCH 115/227] i40e/i40evf: Add new HW link info variable an_enabled
 and function update_link_info
Patch-mainline: v3.17-rc1
Git-commit: 8109e1232b3e5322415a9b5e09951617c5fae277 (partial)
References: bsc#922853 FATE#318529

Add a new variable, hw.phy.link_info.an_enabled, to track whether autoneg is
enabled.  Also add a new function update_link_info that will update that
variable as well as calling get_link_info to update the rest of the link info.
Also add get_phy_capabilities to support this.

Change-ID: I5157ef03492b6dd8ec5e608ba0cf9b0db9c01710
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/intel/i40evf/i40e_type.h |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/intel/i40evf/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40evf/i40e_type.h
@@ -166,6 +166,7 @@ struct i40e_link_status {
 	u8 an_info;
 	u8 ext_info;
 	u8 loopback;
+	bool an_enabled;
 	/* is Link Status Event notification to SW enabled */
 	bool lse_enable;
 	u16 max_frame_size;
